### ISSUE TITLE
I use Intel 710 NIC, but the app can't send packet, I don't know why. :(

### ISSUE THREAD

**Issue by jeremyyx** [stanford-mast/shinjuku#4]
[syx@r740 client]$ ./latency_client 192.170.0.1 1234 1000 100
^C
packets_sent: 2297
packets_received: 0

**Comment by kkaffes** [comment]
Try using the i40e branch or the main branch with this change: https://github.com/stanford-mast/shinjuku/commit/91ac4d1be935d2327d3a3352161465cf0fcfce72

**Comment by jeremyyx** [comment]
Thank you for your reply, actually I ran with this change, the NIC can't send the packet. So I assign tx_ctx.rdylist = 1; Then I can send the packet.I don't understand why, but you did help me solve the problem. :)

**Comment by jeremyyx** [comment]
And I have another problem,if dispatcher call dune_apic_send_posted_ipi()  , shinjuku will become defunct process...

**Comment by rishabh246** [comment]
@YixinSong-e did you ever manage to fix this? I ran into the same issue

**Comment by jeremyyx** [comment]
> @YixinSong-e did you ever manage to fix this? I ran into the same issue

yes, I change the code in dp/drivers/i40e.c 286lines to tx_ctx.rdylist = 1; And it worked.
